### PR TITLE
Add basic benchmarks among engines.

### PR DESCRIPTION
**Pull request by mathetake**
This commit adds the basic benchmark to compare the performance of 
different engine implementations (currently naivevm vs wazeroir interpreter).
Notably this proves that wazeroir interpreter drastically outperforms the 
naivevm by the order of magnitudes. 

The follow up commits will add more variety of cases.

____

```
goos: linux
goarch: amd64
pkg: github.com/tetratelabs/wazero/bench
cpu: AMD Ryzen 9 3950X 16-Core Processor
BenchmarkEngines
BenchmarkEngines/naivevm
BenchmarkEngines/naivevm/base64_5_per_exec
BenchmarkEngines/naivevm/base64_5_per_exec-32         	  399853	      2542 ns/op	     848 B/op	      37 allocs/op
BenchmarkEngines/naivevm/base64_100_per_exec
BenchmarkEngines/naivevm/base64_100_per_exec-32       	  470588	      2443 ns/op	     848 B/op	      37 allocs/op
BenchmarkEngines/naivevm/base64_10000_per_exec
BenchmarkEngines/naivevm/base64_10000_per_exec-32     	  458335	      2532 ns/op	     848 B/op	      37 allocs/op
BenchmarkEngines/naivevm/fibo_for_5
BenchmarkEngines/naivevm/fibo_for_5-32                	   55761	     22410 ns/op	    9224 B/op	     380 allocs/op
BenchmarkEngines/naivevm/fibo_for_10
BenchmarkEngines/naivevm/fibo_for_10-32               	    3958	    266217 ns/op	  108937 B/op	    4511 allocs/op
BenchmarkEngines/naivevm/fibo_for_20
BenchmarkEngines/naivevm/fibo_for_20-32               	      36	  32989618 ns/op	13474027 B/op	  558218 allocs/op
BenchmarkEngines/wazeroir_interpreter
BenchmarkEngines/wazeroir_interpreter/base64_5_per_exec
BenchmarkEngines/wazeroir_interpreter/base64_5_per_exec-32         	 3180720	       365.5 ns/op	     104 B/op	       5 allocs/op
BenchmarkEngines/wazeroir_interpreter/base64_100_per_exec
BenchmarkEngines/wazeroir_interpreter/base64_100_per_exec-32       	 3351330	       360.7 ns/op	     104 B/op	       5 allocs/op
BenchmarkEngines/wazeroir_interpreter/base64_10000_per_exec
BenchmarkEngines/wazeroir_interpreter/base64_10000_per_exec-32     	 3068493	       371.8 ns/op	     104 B/op	       5 allocs/op
BenchmarkEngines/wazeroir_interpreter/fibo_for_5
BenchmarkEngines/wazeroir_interpreter/fibo_for_5-32                	 3225747	       385.7 ns/op	     120 B/op	       5 allocs/op
BenchmarkEngines/wazeroir_interpreter/fibo_for_10
BenchmarkEngines/wazeroir_interpreter/fibo_for_10-32               	 3055983	       383.7 ns/op	     120 B/op	       5 allocs/op
BenchmarkEngines/wazeroir_interpreter/fibo_for_20
BenchmarkEngines/wazeroir_interpreter/fibo_for_20-32               	 3120697	       383.5 ns/op	     120 B/op	       5 allocs/op
PASS
ok  	github.com/tetratelabs/wazero/bench	16.898s
```

### CODE REVIEW

File: .github/workflows/commit.yaml [hunk 1]
```diff
@@ -62,3 +62,25 @@ jobs:
...
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+
+      - run: make bench
+
+      # TODO: publish results as PR comments if becomes slow?
```
Comment (codefromthecrypt) [inline .github/workflows/commit.yaml:86]: good TODO, though not sure how consistent CI perf is to action that!

File: bench/bench._test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package bench
```
Comment (codefromthecrypt) [inline bench/bench._test.go:1]: is the filename intentionally `bench._test.go`?

File: bench/bench._test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+func BenchmarkEngines(b *testing.B) {
+	buf, _ := os.ReadFile("case/case.wasm")
+	mod, _ := wasm.DecodeModule((buf))
+	b.Run("naivevm", func(b *testing.B) {
+		store := newStore(naivevm.NewEngine())
+		_ = store.Instantiate(mod, "test")
+		runBase64Benches(b, store)
+		runFibBenches(b, store)
+	})
+	b.Run("wazeroir_interpreter", func(b *testing.B) {
```
Comment (codefromthecrypt) [inline bench/bench._test.go:26]: maybe leave out _interpreter until there's an option? that would match `wazeroir.NewEngine()` more cleanly and likely lead to aligned tabs while this coexists with naievevm

File: bench/bench._test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	"testing"
+
+	"github.com/tetratelabs/wazero/wasi"
+	"github.com/tetratelabs/wazero/wasm"
+	"github.com/tetratelabs/wazero/wasm/naivevm"
+	"github.com/tetratelabs/wazero/wasm/wazeroir"
+)
+
+func BenchmarkEngines(b *testing.B) {
+	buf, _ := os.ReadFile("case/case.wasm")
```
Comment (codefromthecrypt) [inline bench/bench._test.go:18]: later we can think of a better name as it feels a placeholder otherwise for now (case/case.go|wasm)

**Review by codefromthecrypt — APPROVED** [review]
Thanks for putting in the scaffolding!

File: .github/workflows/commit.yaml (hunk 1 above)
Comment (mathetake) [reply .github/workflows/commit.yaml:86]: yeah agreed :D remove the comment for now :D

File: bench/bench._test.go (hunk 2 above)
Comment (mathetake) [reply bench/bench._test.go:1]: ooops, not intentional 😄  Thanks1

File: bench/bench._test.go (hunk 3 above)
Comment (mathetake) [reply bench/bench._test.go:26]: agreed!

File: bench/bench._test.go (hunk 4 above)
Comment (mathetake) [reply bench/bench._test.go:18]: 👍

**Comment by mathetake** [comment]
@codefromthecrypt thank you for the review! :D

File: bench/bench_test.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/tetratelabs/wazero/wasi"
+	"github.com/tetratelabs/wazero/wasm"
+	"github.com/tetratelabs/wazero/wasm/naivevm"
+	"github.com/tetratelabs/wazero/wasm/wazeroir"
+)
+
+func BenchmarkEngines(b *testing.B) {
+	buf, _ := os.ReadFile("case/case.wasm")
+	mod, _ := wasm.DecodeModule((buf))
```
Comment (codefromthecrypt) [inline bench/bench_test.go:19]: another bench we can consider is decoding and instantiating a module, as this could affect startup time

Comment (mathetake) [reply bench/bench_test.go:19]: DecodeModule doesn't depend on engines, but we should measure `store.Instantiate` where we actually compile functions using engines. Let's do later.

**Pull request merged**